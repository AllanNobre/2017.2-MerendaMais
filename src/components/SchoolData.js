import React from 'react';
import { View, Text } from 'react-native';

const SchoolData = school => (
  <View>
    <Text>
      Nome Escola: {school.schoolName}
    </Text>
    <Text>
      Email: {school.schoolEmail}
    </Text>
    <Text>
      Telefone: {school.schoolPhone}
    </Text>
  </View>
);

export default SchoolData;
