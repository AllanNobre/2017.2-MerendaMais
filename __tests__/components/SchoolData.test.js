import 'react-native';
import { createSerializer } from 'enzyme-to-json';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SchoolData from '../../src/components/SchoolData';

expect.addSnapshotSerializer(createSerializer());
Enzyme.configure({ adapter: new Adapter() });

const props = {
  school: {
    schoolName: 'Escola Teste',
    schoolEmail: 'teste@teste.com',
    schoolPhone: '9999999999',
  },
};

describe('Testing SchoolData component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SchoolData />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Render props', () => {
    const wrapper = shallow(
      <SchoolData {...props} />,
    );
    expect(wrapper.prop('school')).toMatchSnapshot(props.school);
  });
});
