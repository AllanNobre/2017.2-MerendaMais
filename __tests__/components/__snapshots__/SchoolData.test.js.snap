// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `undefined`;

exports[`Testing SchoolData component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "borderColor": "black",
      "borderRadius": 3,
      "borderWidth": 1,
      "flex": 1,
      "justifyContent": "space-between",
      "marginHorizontal": 15,
      "marginVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingLeft": 2,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "justifyContent": "center",
          "marginHorizontal": 15,
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      ESCOLA SELECIONADA
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Nome da escola: 
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Email: 
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Telefone: 
      </Text>
    </Text>
  </View>
</View>
`;
